addons/sale_project/models/project.py

```diff
@@ -394,20 +409,83 @@ def _get_revenues_items_from_sol(self, domain=None, with_action=True):
                             'type': 'object',
                             'args': json.dumps(args),
                         }
-                        other_revenues['action'] = action_params
+                        if len(sale_order_items) == 1:
+                            action_params['res_id'] = sale_order_items.id
+                        materials['action'] = action_params
         sequence_per_invoice_type = self._get_profitability_sequence_per_invoice_type()
         return {
             'data': [{'id': invoice_type, 'sequence': sequence_per_invoice_type[invoice_type], **vals} for invoice_type, vals in revenues_dict.items()],
             'total': {'to_invoice': total_to_invoice, 'invoiced': total_invoiced},
         }
 
+    def _get_aal_domain(self):
+        """
+            Taking only the aal where the sale_line_ids is not enough, for some invoices are related to the project, but not the sales order.
+            Those invoices are already computed in the revenues section, and therefore should not be taken into account. The only way to sort the invoices out
+            is with the move lines.
+        """
+        move_line_ids_no_sol = list(self.env['account.move.line']._search([('sale_line_ids', '=', False)]))
+        return [('account_id', '=', self.analytic_account_id.id), '|', ('move_line_id', '=', False), ('move_line_id', 'in', move_line_ids_no_sol)]
```

Comment:
```suggestion
        return [('account_id', '=', self.analytic_account_id.id), ('move_line_id', 'in', move_line_ids_no_sol + [False])]
```
Is the same. Up to you!